### PR TITLE
Sketch the Kysely adapter for sqld

### PR DESCRIPTION
**Pull request by honzasp**
Using the Hrana client from `@libsql/hrana-client`, this PR sketches an adapter ("dialect") for Kysely.

### CODE REVIEW

File: packages/kysely-sqld/package.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  "name": "@libsql/kysely-sqld",
+
+  "type": "module",
+  "main": "lib-cjs/index.js",
+  "types": "lib-esm/index.d.ts",
+  "exports": {
+    ".": {
+      "types": "./lib-esm/index.d.ts",
+      "import": "./lib-esm/index.js",
+      "require": "./lib-cjs/index.js"
```
Comment (MarinPostma) [inline packages/kysely-sqld/package.json:11]: I don't get why sometimes things come from the `lib-esm` dir, sometimes, from the `lib-cjs`?

**Review by MarinPostma — APPROVED** [review]
looks good

File: packages/kysely-sqld/package.json (hunk 1 above)
Comment (honzasp) [reply packages/kysely-sqld/package.json:11]: ES modules come from `lib-esm`, and CJS modules come from `lib-cjs`. TypeScript declarations are the same regardless of the target module system, so they are produced only in `lib-esm`.

Comment (honzasp) [reply packages/kysely-sqld/package.json:11]: They are emitted here by the two invocations of `tsc` from `scripts.build-{cjs,esm}`, using `tsconfig.build-{cjs,esm}.json` as tsconfig.

**Comment by honzasp** [comment]
@penberg You are right, thanks for pointing out, fixed!

**Comment by MarinPostma** [comment]
bors merge

**Pull request merged**